integration/framework/scenario.go

```diff
@@ -0,0 +1,292 @@
+package framework
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+
+	"github.com/pkg/errors"
+)
+
+var (
+	nextScenarioID = 1
+)
+
+type Scenario struct {
+	id          int
+	networkName string
+	services    []*Service
+}
+
+func NewScenario() (*Scenario, error) {
+	s := &Scenario{
+		id:          nextScenarioID,
+		networkName: "cortex-" + strconv.Itoa(nextScenarioID),
+		services:    []*Service{},
+	}
+
+	// Setup the docker network
+	if out, err := RunCommandAndGetOutput("docker", "network", "create", s.networkName); err != nil {
```

Comment:
The problem is not removing or reusing the network if already exists, but after that error you will get an error for already existing containers. You get this error if the previous test run panicked (ie. test timeout) and automatically cleaning up containers in a safe way is non trivial.

Please let me know if you ended up into this situation without a panic.

Reply:
It was pretty common to run into problem with container left behind, when I didn't have CORTEX_IMAGE and  CORTEX_INTEGRATION_DIR set correctly.

With unset CORTEX_INTEGRATION_DIR, docker starts minio container, but minio then fails. However, since this is reported as error in the `StartService` code, service is not appended to list of started services, and is then not cleaned up.

Reply:
After talking to Peter, we actually found an issue. If `service.Start()` did return `error` (for any reason) the container may have been created but it was not cleaned up. I've pushed a fix for this.

Reply:
In the previous version I force-removed all containers and the network before running each scenario.
(That is what `stop-components.sh` did)

Reply:
Ok. I've added the logic to cleanup the Docker containers + network before each test, so that we can now automatically recover from a spurious situation left from the previous run. The change is in [this commit](https://github.com/cortexproject/cortex/pull/1839/commits/f81d2825a8c723902c1bdadef639cfa03afd2b54).